idaes/core/util/tables.py

```diff
@@ -91,7 +91,10 @@ def _stream_dict_add(sb, n, i=None):
     for n in streams.keys():
         if isinstance(streams[n], Arc):
             for i, a in streams[n].items():
-                sb = _get_state_from_port(a.ports[1], time_point)
+                try:
+                    sb = _get_state_from_port(a.ports[1], time_point)
+                except:
+                    sb = _get_state_from_port(a.ports[0], time_point)
```

Comment:
Does this try-except block exist to deal with input ports to surrogate models? The general philosophy has been "input ports have well-defined states, output ports may not", so putting in a comment explaining what the use-case is would be good.

Reply:
I'll add a comment.  In this case, it deals with me making an occasional translator port or having a unit without state blocks.  It just makes it more likely I'll actually find a state block.  Yeah, I guess the same would apply to surrogate models.

Reply:
So, a general comment on surrogates, but with the proposed new API, whilst there will not be a `StateBlock` present, all the variables *should* be part of the same `Block`. So, `_get_state_from_port` should return a block for most surrogates, but it would be a `SurrogateBlock` or `SkeletonModel` instead of a `StateBlock`.

Reply:
@andrewlee94, I agree with that.  It may require a little more from the modeler, but any block with the standard property names for whatever is available would be good.  Then we just attach it as a new attribute onto the port maybe.  So IDAES ports should be expected to have a state_block attribute.  For the future though.

Reply:
It would be good, eventually, to add in code such that specific exceptions are being caught and if `_get_state_from_port` fails for both ports an error message would say so, rather than that it just couldn't find a state block on the outlet port. Probably beyond the scope of this changeset, though.